packages/cache/README.md

```diff
@@ -0,0 +1,85 @@
+# Agoric Cache
+
+This cache mechanism allows a cache client function to synchronize with a cache
+backend.  Any passable object can be a cache key or a cache value.
+
+## Demo
+
+```js
+import { makeCache, makeStoreCoordinator } from '@agoric/cache';
+import { makeScalarBigMapStore } from '@agoric/vat-data';
+
+const store = makeScalarBigMapStore('cache');
+const coordinator = makeStoreCoordinator(store);
+const cache = makeCache(coordinator);
+
+// Direct value manipulation.
+await cache('baz'); // undefined
+await cache('baz', 'barbosa'); // 'barbosa'
+
+// Match-and-set.
+await cache('baz', 'babaloo', undefined); // 'barbosa'
+await cache('baz', 'babaloo', 'barbosa'); // 'babaloo'
+
+
+// Update the `'foo'` entry, using its old value (initially `undefined`).
+await cache('foo'); // `undefined`
+const updater = (oldValue = 'bar') => `${oldValue}1`;
+await cache('foo', updater); // 'bar1'
+await cache('foo', updater); // 'bar11'
+await cache('foo'); // 'bar11'
+
+// You can also assert a pattern for the key.  If it doesn't match, then the cache returns a rejection.
+await cache('foo', updater, 'nomatch'); // 'bar11'
+await cache('foo', updater, 'bar11'); // 'bar111'
+await cache('foo', updater, 'bar11'); // 'bar111'
+await cache('foo'); // 'bar111'
+
+// Specify a pattern of `undefined` for one-time initialisation.
+await cache('frotz', 'default', undefined); // 'default'
+await cache('frotz', 'ignored', undefined); // 'ignored'
+```
+
+## Cache client
+
+The client-side API is normally exposed as a single function named `cache`.  You
+can create a cache client function by running `makeCache(coordinator,
+follower)`.  If not specified, the default coordinator is just a local in-memory
+map without persistence.
+
+- the ground state for a cache key value is `undefined`.  It is impossible to distinguish a set value of `undefined` from an unset key
+- `cache(key): Promise<recentValue>` - get an eventually-consistent value for `key`
+- `cache(key, (oldValue) => ERef<newValue>): Promise<newValue>` -  call the updater function transactionally with the current value of `key`, and update it to `newValue`.  Rerun the updater function with a new value if the transaction is  stale.
```

Comment:
Super nit: extra space before "stale". Is it also convention to have two spaces after a period?

Reply:
Yes, two spaces after a period is stylish.